packages/apollo-server-fastify/README.md

```diff
@@ -0,0 +1,45 @@
+---
+title: Fastify
+description: Setting up Apollo Server with Fastify
+---
+
+[![npm version](https://badge.fury.io/js/apollo-server-fastify.svg)](https://badge.fury.io/js/apollo-server-fastify) [![Build Status](https://circleci.com/gh/apollographql/apollo-server.svg?style=svg)](https://circleci.com/gh/apollographql/apollo-server) [![Coverage Status](https://coveralls.io/repos/github/apollographql/apollo-server/badge.svg?branch=master)](https://coveralls.io/github/apollographql/apollo-server?branch=master) [![Get on Slack](https://img.shields.io/badge/slack-join-orange.svg)](https://www.apollographql.com/#slack)
+
+This is the Fastify integration of GraphQL Server. Apollo Server is a community-maintained open-source GraphQL server that works with many Node.js HTTP server frameworks. [Read the docs](https://www.apollographql.com/docs/apollo-server/). [Read the CHANGELOG.](https://github.com/apollographql/apollo-server/blob/master/CHANGELOG.md)
+
+```sh
+npm install apollo-server-fastify
+```
+
+## Fastify
+
+```js
+const { ApolloServer, gql } = require('apollo-server-fastify');
+const fastify = require('fastify');
+
+async function StartServer() {
+  const server = new ApolloServer({ typeDefs, resolvers });
+
+  const app = fastify();
+
+  await server.applyMiddleware({
+    app,
+  });
+
+  await server.installSubscriptionHandlers(app.server);
+
+  await app.listen(3000);
+}
```

Comment:
The best approach would be:

```js
const { apolloFastify, gql } = require('apollo-server-fastify');
const fastify = require('fastify');
 async function StartServer() {
   const app = fastify();
  
   // apolloFastify is a fastify plugin
   // it wraps  ApolloServer
   app.register(apolloFastify, {
    typeDefs, resolvers 
  })

   await app.listen(3000);
}
```

Reply:
Makes sense, I was trying to keep the interface the same as server integrations but I guess we can do both

Reply:
Drive-by comment: I want to be mindful of any limitations which might currently be imposed which would prevent a ubiquitous interface, but we should certainly _try_ to keep the interface the same as the other server integrations for portability and ease of documentation.

Reply:
Overall, there are no big limitation to keep the previous interface. However, wrap everything in a plugin inside that method.

This API would look very alien for a fastify user.

Reply:
I think what I'm getting at is that the `createHandler` technique also employed by other Apollo Server integrations.  e.g.:

* [`apollo-server-azure-functions`](https://www.npmjs.com/package/apollo-server-azure-functions)
* [`apollo-server-cloud-functions`](https://www.npmjs.com/package/apollo-server-cloud-functions)
* [`apollo-server-lambda`](https://www.npmjs.com/package/apollo-server-lambda)

Therefore, it might/should look like:

```js
const { ApolloServer, gql } = require('apollo-server-fastify');
const { typeDefs, resolvers } = require('./module');

const server = new ApolloServer({
  typeDefs,
  resolvers,
});

const app = require('fastify')();

(async function () { 
  app.register(server.createHandler());
  await app.listen(3000);
})();
```

Would this be foreign to Fastify users?

Reply:
Definitely +1 on this.

Reply:
Sounds like we have a solution, great! Thanks for the input